integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonMergerRDD.scala

```diff
@@ -359,6 +359,8 @@ class CarbonMergerRDD[K, V](
       loadMetadataDetails = SegmentStatusManager
         .readLoadMetadata(CarbonTablePath.getMetadataPath(tablePath))
     }
+
+    val validSegIds: java.util.List[String] = new util.ArrayList[String]()
     // for each valid segment.
     for (eachSeg <- carbonMergerMapping.validSegments) {
       // In case of range column get the size for calculation of number of ranges
```

Comment:
This loop can be optimized, `if (null != rangeColumn)` is not changing in the loop

Reply:
`if(null != rangeColumn)` check is for calculating the total size for range column based on valid segments in the loop. so, i think it cannot be optimized further